src/if.js

```diff
@@ -43,20 +43,42 @@ export class If {
   * @param newValue The new value
   */
   valueChanged(newValue) {
+    if (this.__queuedChanges) {      
+      this.__queuedChanges.push(newValue);
+      return;
+    }
+
+    let maybePromise = this._runValueChanged(newValue);
+    if (maybePromise instanceof Promise) {
+      let queuedChanges = this.__queuedChanges = [];
+
+      let runQueuedChanges = () => {
+        if (!queuedChanges.length) {
+          delete this.__queuedChanges;       
+          return;
+        }        
+        let nextPromise = this._runValueChanged(queuedChanges.shift()) || Promise.resolve();          
```

Comment:
@atsu85 If a new change happens before an animation (ie the method `valueChanged`) has finished the new value will be pushed to `__queuedChanges` on line 47 and `queuedChanges.length` will be `> 0`, does that makes sense?